From f42f8050b10ca9e7af1fb644a77acc0d7c624f9d Mon Sep 17 00:00:00 2001
From: sunchao <sunchao106@huawei.com>
Date: Mon, 11 Nov 2024 11:32:29 +0800
Subject: [PATCH] Support the Depth Fusion interface TicketNo:DTS2024111109303
 Description:Support the Depth Fusion interface Team:Gitee Feature or
 Bugfix:Feature Binary Source:Sync from gitee PrivateCode(Yes/No):Yes

Change-Id: I3adca3639ff2b7a791ded3f7f68027832d8f0eac
---
 api/@ohos.multimedia.camera.d.ts | 70 ++++++++++++++++++++++++++++++++
 1 file changed, 70 insertions(+)

diff --git a/api/@ohos.multimedia.camera.d.ts b/api/@ohos.multimedia.camera.d.ts
index 1bf7a2247..27d2f9f86 100644
--- a/api/@ohos.multimedia.camera.d.ts
+++ b/api/@ohos.multimedia.camera.d.ts
@@ -10324,6 +10324,76 @@ declare namespace camera {
      */
     off(type: 'error', callback?: ErrorCallback): void;
   }
+
+  /**
+   * Depth Fusion Query object.
+   *
+   * @interface DepthFusionQuery
+   * @syscap SystemCapability.Multimedia.Camera.Core
+   * @systemapi
+   * @since 14
+   */
+  interface DepthFusionQuery {
+    /**
+     * Checks whether a depth fusion is supported.
+     *
+     * @returns { boolean } Is the depth fusion supported.
+     * @throws { BusinessError } 202 - Not System Application.
+     * @throws { BusinessError } 7400103 - Session not config, only throw in session usage.
+     * @syscap SystemCapability.Multimedia.Camera.Core
+     * @systemapi
+     * @since 14
+     */
+    isDepthFusionSupported(): boolean;
+
+    /**
+     * Query the depth fusion threshold.
+     *
+     * @returns { Array<number> } The depth fusion threshold.
+     * @throws { BusinessError } 202 - Not System Application.
+     * @throws { BusinessError } 7400103 - Session not config, only throw in session usage.
+     * @syscap SystemCapability.Multimedia.Camera.Core
+     * @systemapi
+     * @since 14
+     */
+    getDepthFusionThreshold(): Array<number>;
+  }
+
+  /**
+   * Depth Fusion object.
+   *
+   * @interface DepthFusion
+   * @syscap SystemCapability.Multimedia.Camera.Core
+   * @systemapi
+   * @since 14
+   */
+  interface DepthFusion extends DepthFusionQuery {
+    /**
+     * Confirm if the depth fusion enabled.
+     *
+     * @returns { boolean } TRUE if the depth fusion is enable.
+     * @throws { BusinessError } 202 - Not System Application.
+     * @throws { BusinessError } 7400103 - Session not config.
+     * @syscap SystemCapability.Multimedia.Camera.Core
+     * @systemapi
+     * @since 14
+     */
+    isDepthFusionEnabled(): boolean;
+
+    /**
+     * Enable depth fusion.
+     *
+     * @param { boolean } enabled - Target state for depth fusion.
+     * @throws { BusinessError } 202 - Not System Application.
+     * @throws { BusinessError } 7400101 - Parameter missing or parameter type incorrect.
+     * @throws { BusinessError } 7400103 - Session not config.
+     * @throws { BusinessError } 7400201 - Camera service fatal error.
+     * @syscap SystemCapability.Multimedia.Camera.Core
+     * @systemapi
+     * @since 14
+     */
+    enableDepthFusion(enabled: boolean): void;
+  }
 }
 
 export default camera;
-- 
2.45.2.huawei.6
